PatTools/python/patFinalStateProducers.py

```diff
@@ -131,126 +106,73 @@ def produce_final_states(process, collections, output_commands,
-                src=cms.InputTag("muonsForFinalStates"),
-                algorithm=cms.string("byDeltaR"),
-                preselection=cms.string(
-                    "pt>10&&isGlobalMuon&&isTrackerMuon&&"
-                    "(userIso(0)+max(photonIso()+neutralHadronIso()"
-                    "-0.5*puChargedHadronIso(),0))/pt()<0.3"),
-                deltaR=cms.double(0.3),
-                checkRecoComponents=cms.bool(False),
-                pairCut=cms.string(""),
-                requireNoOverlaps=cms.bool(True),
-            ),
-            electrons=cms.PSet(
-                src=cms.InputTag("electronsForFinalStates"),
-                algorithm=cms.string("byDeltaR"),
-                preselection=cms.string(electronCleaning),
-                deltaR=cms.double(0.3),
-                checkRecoComponents=cms.bool(False),
-                pairCut=cms.string(""),
-                requireNoOverlaps=cms.bool(True),
-            ),
-        ),
-        # finalCut (any string-based cut on pat::Jet)
-        finalCut=cms.string('')
-    )
-    process.jetsForFinalStates = cms.EDProducer(
+        src=cms.InputTag(src['taus']))
+    src['taus'] = 'tausRank'
+
+    process.jetsRank = cms.EDProducer(
         "PATJetRanker",
-        src=cms.InputTag("jetsFiltered"))
+        src=cms.InputTag(src['jets']))
+    src['jets'] = 'jetsRank'
 
-    process.selectObjectsForFinalStates = cms.Sequence(
+    process.rankSequence = cms.Sequence(
         process.muonsRank
-        + process.muonsForFinalStates
         + process.electronsRank
-        + process.electronsForFinalStates
         + process.tausRank
-        + process.tausForFinalStates
-        + process.jetsFiltered
-        + process.jetsForFinalStates
-    )
-    if not noPhotons:
-        process.selectObjectsForFinalStates += process.photonsForFinalStates
+        + process.jetsRank
+        )
+    process.rankObjects = cms.Path(process.rankSequence)
+    process.schedule.append(process.rankObjects)
 
-    sequence += process.selectObjectsForFinalStates
 
-    # Now build all combinatorics for E/Mu/Tau/Photon
-    object_types = [('Elec', cms.InputTag("electronsForFinalStates")),
-                    ('Mu', cms.InputTag("muonsForFinalStates")),
-                    ('Tau', cms.InputTag("tausForFinalStates")),
-                    ('Jet', cms.InputTag("jetsForFinalStates"))]
+    ### apply final selections to the objects we'll use in the final states
+    # Initialize final-state object sequence
+    finalSelections = kwargs.get('finalSelection',{})
+    from FinalStateAnalysis.NtupleTools.object_parameter_selector import setup_selections, getName
+    process.selectObjectsForFinalStates = setup_selections(
+        process, 
+        "FinalSelection",
+        src,
+        finalSelections,
+        )
+    for ob in finalSelections:
+        src[getName(ob)+"s"] = getName(ob)+"FinalSelection"
+        output_commands.append('*_%sFinalSelection_*_*'%getName(ob))
```

Comment:
Ah crap, I didn't read carefully enough. I didn't realize this was embedding a variable. I'll do it after the cleaning.